src/WingetCreateCore/Common/GitHub.cs

```diff
@@ -200,6 +200,17 @@ public async Task<string> GetFileContentsAsync(string path)
         public async Task CheckAccess()
         {
             await this.github.Repository.Get(this.wingetRepoOwner, this.wingetRepo);
+        }
+
+        /// <summary>
+        /// Recursively searches the repository for the provided package identifer to determine if it already exists.
+        /// </summary>
+        /// <param name="packageId">The package identifier.</param>
+        /// <returns>Boolean value indicating if a duplicate manifest exists.</returns>
+        public async Task<bool> CheckDuplicatePackageId(string packageId)

```

Comment:
>CheckDuplicatePackageId [](http://example.com/codeflow?start=32&length=23)

This will return true even if the packageId is only a substring of something in the repo, for example, microsoft.visualstudio.2019 returns true.

I think this is probably fine, as I can't imagine we'd want to allow a package in that's just a substring of another one, but @denelon to confirm

Reply:
Take a look at Visual Studio Code, and Visual Studio Code Insiders.

Reply:
In my opinion, I think this is okay. Even if the inputted packageId is only a substring, that still is something we shouldn't allow because it is matching an existing directory. We want to ensure that the new packageId the user is creating is completely unique.